presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestTimestampCompatibility.java

```diff
@@ -104,4 +105,123 @@ public void testTimestampWithNullCompatibility(String storageFormat, String lega
 
         resetSessionProperty(connection, "legacy_timestamp");
     }
+
+    @Test(dataProvider = "storage_formats", groups = {HIVE_CONNECTOR, TIMESTAMP})
+    public void testSingleRowArrayCompatibility(String storageFormat, String legacyTimestamp)
+            throws SQLException
+    {
+        if (storageFormat.equals("PARQUET")) {
+            throw new SkipException("This is disabled due to #8729 bug in Presto.");
+        }
+        Connection connection = defaultQueryExecutor().getConnection();
+        setSessionProperty(connection, "legacy_timestamp", legacyTimestamp);
+
+        query(String.format("DROP TABLE IF EXISTS %s", TABLE_NAME));
+        query(String.format("CREATE TABLE %s WITH (format = '%s') AS SELECT ARRAY[TIMESTAMP '%s', TIMESTAMP '%s'] ts", TABLE_NAME, storageFormat, TIMESTAMP_LITERAL, TIMESTAMP_LITERAL));
+
+        QueryResult prestoResult = query(String.format("SELECT ts[1], ts[2] FROM %s", TABLE_NAME));
+        QueryResult hiveResult = onHive().executeQuery(String.format("SELECT ts[0], ts[1] FROM %s", TABLE_NAME));
+        assertThat(hiveResult).containsExactly(row(EXPECTED_TIMESTAMP, EXPECTED_TIMESTAMP));
+        assertThat(prestoResult).containsExactly(row(EXPECTED_TIMESTAMP, EXPECTED_TIMESTAMP));
+
+        resetSessionProperty(connection, "legacy_timestamp");
+    }
+
+    @Test(dataProvider = "storage_formats", groups = {HIVE_CONNECTOR, TIMESTAMP})
+    public void testNullInArrayCompatibility(String storageFormat, String legacyTimestamp)
```

Comment:
maybe you could merge this test with the test above, so we could omit 12 tests execution. then above tests would check timestamp and null values together (two rows vs single one) 

EDITED: as I read more tests it looks like we could merge them all. Currently there is a lot copy-paste